randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -20,7 +20,14 @@ var WallpaperController = class {
 
 	constructor() {
 		this.logger = new LoggerModule.Logger('RWG3', 'WallpaperController');
-		this.wallpaperlocation = Self.path + '/wallpapers/';
+		let xdg_cache_home = Mainloop.getenv('XDG_CACHE_HOME')
+		if (!xdg_cache_home)
+		{
+			xdg_cache_home = Mainloop.getenv('HOME') + '/.cache'
+		}
+		this.wallpaperlocation = xdg_cache_home + '/RandomWallpaperGnome3/wallpapers/';
```

Comment:
I think it would be better to use `Self.metadata['uuid']` instead of the hardcoded "RandomWallpaperGnome3" as directory here.
Additionally, if you feel like it, you could use JS string interpolation instead of string concatenation but I'll leave that up to you.